### PR TITLE
Expose host settings during app:add and include sane defaults.

### PR DESCRIPTION
**Pull request by EclipseGc**

### CODE REVIEW

**Comment by jonpugh** [comment]
In my experience building devshop, the "host" is really tied to the environment, as the "app" represents just the source code itself, before it is running anywhere. However, the "app" should have a "default host" setting that environments can use when they are created.  

This way when you create an environment, you don't have to choose what the host will be, but you can still override it if you want to.

Aside:

"Servers", "hosts", or "providers", whatever we want to call them, should become a real entity at some point. Tracking the status of the machines running our apps is in the scope of this project.

We should handle "servers" as we do with aegir: a hostname and an IP, with a single default server created on installation.

server created as a default, but that can be for the next PR.

**Comment by EclipseGc** [comment]
So you're asking me to take this a step further and provide environment level host overrides? I'm ++ to that notion, but I think we have to start tracking "Servers", "hosts", "providers" first. I think we're overdue for that, but this patch is a good "hobble-along" until we restructure that more permanently.

Thoughts?

File: src/terra/Factory/EnvironmentFactory.php [hunk 1]
```diff
@@ -75,9 +75,11 @@ public function init($path = null)
         }
 
         try {
+            mkdir($path, 0755, TRUE);
+            chdir($path);
             $wrapper = new GitWrapper();
             $wrapper->streamOutput();
-            $wrapper->clone($this->app->repo, $path);
+            $wrapper->cloneRepository($this->app->repo . '.git', $path);
```
Comment (jonpugh) [inline src/terra/Factory/EnvironmentFactory.php:82]: @EclipseGc What if the repo includes the `.git` at the end? wouldn't this break?

Comment (EclipseGc) [inline src/terra/Factory/EnvironmentFactory.php:82]: Yes it absolutely would, we should probably do some sort of cli level validation of the input for the app:add command.

**Comment by jonpugh** [comment]
Works great, thanks!

**Pull request merged**